### PR TITLE
[FIXED JENKINS-45235] When multiple update sites publish different versions, allow those versions to be seen

### PR DESCRIPTION
**Pull request by stephenc**
- When automatically installing dependent plugins, the first update center with a version meeting the minimum required will win
- Also enhances the CLI command to allow specification of a minimum version

See [JENKINS-45235](https://issues.jenkins-ci.org/browse/JENKINS-45235).

### Proposed changelog entries

* When Jenkins has multiple update sites, the plugins listed were de-duplicated, however this could hide updated versions being published by the second or subsequent update sites, now multiple versions of plugins will be shown when the update sites are publishing different versions.

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
    * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
    * Very tricky to write a test for this, requires multiple update sites with different plugin versions. Manually verified the functionality. Existing test cases should cover the single update site case for the 90%+ of users
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@daniel-beck this should fix the issue I described and help testing of plugin security releases

### CODE REVIEW

File: core/src/main/java/hudson/model/UpdateCenter.java [hunk 1]
```diff
@@ -615,6 +616,19 @@ public String getDefaultBaseUrl() {
         return null;
     }
 
+    /**
+     * Gets the plugin with the given name from the first {@link UpdateSite} to contain it.
+     * @return Discovered {@link Plugin}. {@code null} if it cannot be found
+     */
+    public @CheckForNull Plugin getPlugin(String artifactId, VersionNumber minVersion) {
+        for (UpdateSite s : sites) {
```
Comment (dongsik2026) [inline core/src/main/java/hudson/model/UpdateCenter.java]: Good use of foreach statement!

File: core/src/main/resources/hudson/PluginManager/_table.js [hunk 2]
```diff
@@ -28,8 +28,11 @@ Behaviour.specify("#filter-box", '_table', 0, function(e) {
             var items = document.getElementsBySelector(clz);
             for (var i=0; i<items.length; i++) {
                 var visible = (filter=="" || items[i].innerHTML.toLowerCase().indexOf(filter)>=0);
-                var name = items[i].getAttribute("name");
+                var name = items[i].cells && items[i].cells.length > 1
+                        ? items[i].cells[1].getAttribute('data-id')
+                        : items[i].getAttribute("name");
                 if (visible && name != null) {
+                    console.log(name);
```
Comment (dongsik2026) [inline core/src/main/resources/hudson/PluginManager/_table.js]: why do not use the logger?

**Review by dongsik2026 — CHANGES_REQUESTED** [review]

**Comment by daniel-beck** [comment]
Tests report NPEs in `UpdateCenter.getPlugin`

**Comment by stephenc** [comment]
@daniel-beck so remaining test failure seems unrelated: a job rename ath only on firefox

**Pull request merged**